r2dbc-mysql/src/main/java/SimpleStatement.kt

```diff
@@ -45,9 +47,40 @@ class SimpleStatement(private val clientSupplier: Supplier<JasyncConnection>, pr
         return this
     }
 
+    override fun returnGeneratedValues(vararg columns: String): Statement {
+        if (columns.any { !it.equals("LAST_INSERT_ID", true) }) {
```

Comment:
The intention of `columns` is to specify a desired column name for generated keys. So the first column name should be used as an alias (`SELECT LAST_INSERT_ID() as $columns[0]`). See [SQL Server implementation](https://github.com/r2dbc/r2dbc-mssql/blob/master/src/main/java/io/r2dbc/mssql/GeneratedValues.java#L112) for further details.

Reply:
That can be simplified to:

```kotlin
        if (columns.size == 1) {
            generatedKeyName = columns[0]
        }

        if (columns.size > 1) {
            throw IllegalArgumentException("MySQL only supports a single generated value")
        }

        selectLastInsertId = true

        return this
```